### DISCUSSION TITLE
How to use it in a workspace while the packages has dependent relationship

### DISCUSSION THREAD

**Discussion by chiichen** [release-plz/release-plz#1664] [Q&A] [answered]
I'm using github action in repo https://github.com/Chiichen/libvnc-rs. Including `libvnc` and `libvnc-sys`. It's a crate providing bindings to  https://github.com/LibVNC/libvncserver.`libvnc-sys` provides a low-level binding to the origin library. And `libvnc` is a high-level bingding  based on `libvnc-sys`. So it's obviously that the version of `libvnc` is related with the version of `libvnc-sys`

When the version of `libvnc-sys` is modified(eg. 0.1.0->0.1.1), I want to :
1. Bump `libvnc-sys` version in Cargo.toml (libvnc-sys)
2. Bump `libvnc-sys` dependency version in Cargo.toml(libvnc)
3. Bump `libvnc` version in Cargo.toml

Besides, we need to publish the libvnc-sys and libvnc seperately. So I'm looking for a way to generate two release PR (one for libvnc-sys and another for libvnc)

**Comment by chiichen** [comment]
I figured out a way to skip step2. I use `libvnc-sys = "0.1"` to use the latest version(Though it's not the best solution. We need to bump the major version manually either). So how to make sure publish `libvnc-sys` first and then publish `libvnc` is all I need.

**Comment by marcoieni** [comment]
Would https://github.com/MarcoIeni/release-plz/issues/1029 fix all your issues?

**Reply by chiichen** [reply]
> #1029能解决您的所有问题吗？

I think it can resolve most of my issue except of `upgrading corresponding dependencies in workspace`. If I bumped package `A` from 0.1.0 to 0.1.1. I need the action to do :
1. Bump version of A from 0.1.0 to 0.1.1 and release it to crate.io
2. If 1 success, update all the version of dependency A in workspace from 0.1.0 to 0.1.1

**Reply by marcoieni** [reply]
> upgrading corresponding dependencies in workspace

doesn't release-plz does this already?

If package A changes, release-plz bumps all packages that depend on package B.

You can test it easily locally by running `release-plz update` after you add a commit to package A

**Reply by chiichen** [reply, marked as answer]
> > upgrading corresponding dependencies in workspace
> 
> doesn't release-plz does this already?
> 
> If package A changes, release-plz bumps all packages that depend on package B.
> 
> You can test it easily locally by running `release-plz update` after you add a commit to package A

I figured out the reason.It looks like I need to specify the `path` in dependency explicitly to make it works.By the way, if there is a dependency relationship like package `A` depends on package `B`, does release-plz action ensure publish package `B` first?

**Reply by marcoieni** [reply]
> It looks like I need to specify the path in dependency explicitly to make it works

yes! 👍 

> By the way, if there is a dependency relationship like package A depends on package B, does release-plz action ensure publish package B first?

of course 👍 

Thank I'm marking this discussion as solved since the open issue should track your request. If you want to work on that issue, pr welcome. I don't have the time to work on it right now.
